transport/transport-runtime/src/androidTest/java/com/google/android/datatransport/runtime/SchedulerIntegrationTest.java

```diff
@@ -0,0 +1,180 @@
+            .setPayload("TelemetryData".getBytes())
+            .build();
+    transport.send(stringEvent);
+    verify(mockBackend, times(1)).decorate(eq(expectedEvent));
+    SystemClock.sleep(5000);
+    verify(mockUploader, times(1)).upload(eq(mockBackendName), eq(1), any());
+  }
+
+  @Test
+  public void scheduler_whenEventsScheduledWithSameBackend_shouldUploadOnce() {
+    String mockBackendName = generateBackendName();
+    when(mockRegistry.get(mockBackendName)).thenReturn(mockBackend);
+    TransportFactory factory = runtime.newFactory(mockBackendName);
+    Transport<String> transport =
+        factory.getTransport(testTransport, String.class, String::getBytes);
+    Event<String> stringEvent = Event.ofTelemetry("TelemetryData");
+    EventInternal expectedEvent =
+        EventInternal.builder()
+            .setEventMillis(3)
+            .setUptimeMillis(1)
+            .setTransportName(testTransport)
+            .setPayload("TelemetryData".getBytes())
+            .build();
+    Event<String> stringEvent2 = Event.ofTelemetry("TelemetryData2");
+    EventInternal expectedEvent2 =
+        EventInternal.builder()
+            .setEventMillis(3)
+            .setUptimeMillis(1)
+            .setTransportName(testTransport)
+            .setPayload("TelemetryData2".getBytes())
+            .build();
+    transport.send(stringEvent);
+    transport.send(stringEvent2);
+    verify(mockBackend, times(1)).decorate(eq(expectedEvent));
+    verify(mockBackend, times(1)).decorate(eq(expectedEvent2));
+    SystemClock.sleep(5000);
+    verify(mockUploader, times(1)).upload(eq(mockBackendName), eq(1), any());
+  }
+
+  @Test
+  public void scheduler_whenEventsScheduledWithDifferentBackends_shouldUploadTwice() {
+    String firstBackendName = generateBackendName();
+    String secondBackendName = generateBackendName();
+    when(mockRegistry.get(firstBackendName)).thenReturn(mockBackend);
+    when(mockRegistry.get(secondBackendName)).thenReturn(mockBackend2);
+    TransportFactory factory = runtime.newFactory(firstBackendName);
+    Transport<String> transport =
+        factory.getTransport(testTransport, String.class, String::getBytes);
+    Event<String> stringEvent = Event.ofTelemetry("TelemetryData");
+    EventInternal expectedEvent =
+        EventInternal.builder()
+            .setEventMillis(3)
+            .setUptimeMillis(1)
+            .setTransportName(testTransport)
+            .setPayload("TelemetryData".getBytes())
+            .build();
+    transport.send(stringEvent);
+    TransportFactory factory2 = runtime.newFactory(secondBackendName);
+    Transport<String> transport2 =
+        factory2.getTransport(testTransport, String.class, String::getBytes);
+    transport2.send(stringEvent);
+    verify(mockBackend, times(1)).decorate(eq(expectedEvent));
+    verify(mockBackend2, times(1)).decorate(eq(expectedEvent));
+    SystemClock.sleep(5000);
+    verify(mockUploader, times(2)).upload(any(), eq(1), any());
```

Comment:
let's verify backend names for each call instead of having `times(2)` with `any()`